### PR TITLE
Additional units for unit checker; fixes #4116

### PR DESCRIPTION
**Pull request by rkraneis**
Addresses https://github.com/typetools/checker-framework/issues/4116.

If I read the contribution guideline correctly discussion should happen in this separate pull request against my own master branch.

### CODE REVIEW

**Comment by mernst** [comment]
> Addresses [typetools#4116](https://github.com/typetools/checker-framework/issues/4116).
> 
> If I read the contribution guideline correctly discussion should happen in this separate pull request against my own master branch.

Sorry for not addressing this earlier; it fell through the cracks because it is against your own repository, and because it is marked as a draft.

1. Pull requests should be made against the master branch.  Can you please point me at the misleading documentation so I can fix it?  Sorry for the confusion.

2. Please make a pull request against the master branch. Please don't mark the pull request as a draft, or people will think it should not yet be reviewed.

Thanks!  This will be a nice improvement to merge into the Checker Framework.

**Comment by rkraneis** [comment]
Hello @mernst,
the documentation I tried to follow is https://rawgit.com/typetools/checker-framework/master/docs/developer/developer-manual.html#pull-requests
> It is good style to create a branch (in your fork of the Checker Framework GitHub repository) for each independent change. Do not make changes to your master branch. Do not make changes to a branch of the upstream (typetools) repository. Such a branch competes for CI resources with all pull requests. Make changes on a branch of your own fork, so that both you and others get faster CI feedback. 

I think I by default assumed to just create a PR against my own master branch :thinking: 

Draft isn't mentioned in the docs, I fear I just brought that habit over from work projects. It might be that I read about discussing a proposed feature in the PR and not in an issue in another issue, but fail to find the reference now. The main manual itself only talks about “Contributing fixes (creating a pull request)” (in the [troubleshooting section](https://checkerframework.org/manual/#troubleshooting)); I guess what I want to add in this PR sits somewhere between a bug fix and a feature anyway :)

**Comment by rkraneis** [comment]
Will re-open targeting typetools/checkerframework:master.

**Comment by mernst** [comment]
I [clarified the developer manual](https://github.com/typetools/checker-framework/commit/3f454c84a63549f5b9c2e977d867f0475e62f609); do you think that will help avoid confusion?  Feel free to suggest further wording improvements.

You are right that other projects use draft pull requests for discussions.  We don't have that habit.  I'm not opposed to it on principle, though.  (It is unfortunate that it uses CI resources and thus slows down CI feedback for other pull requests.)

**Comment by rkraneis** [comment]
I think the information that might still missing (or not spelled out explicitly) is where to put the discussion about a proposed feature. Additionally, if I understand you correctly, me having opened a PR against the typetools checker repository will now unduly consume resources on each push while I finish the feature? (Similar to what would happen with a branch not in my but in the checker repo.) That would be quite unfortunate. I will then limit the impact by first pushing to a separate stabilizing branch before updating the PR branch for additional review.
So to summarize: a feature should be developed in a fork and a PR only created after CI (using free azure quota) is green (and the feature really ready to review).

**Comment by mernst** [comment]
All this is correct.  The penultimate paragraph of https://rawgit.com/typetools/checker-framework/master/docs/developer/developer-manual.html#pull-requests now mentions discussions in draft pull requests.

Thanks again.
